### PR TITLE
update pubsub dashboard gateway json panels and remove ELP nodes from dev - PLAT-224

### PR DESCRIPTION
**Pull request by gvelez17**
Adjusted pubsub gateway panels to pull data from correct source

Removed ELP panels from non-prod

https://linear.app/3boxlabs/issue/PLAT-224 - autolink is not in place due to # repos limitation

### CODE REVIEW

File: grafana/provisioning/dashboards/dev/pubsub.json [hunk 1]
```diff
@@ -1,1576 +1,1576 @@
...
-                "builtIn": 1,
-                "datasource": "-- Grafana --",
-                "enable": true,
-                "hide": true,
-                "iconColor": "rgba(0, 211, 255, 1)",
-                "name": "Annotations & Alerts",
-                "type": "dashboard"
-            }
-        ]
+  "annotations": {
```
Comment (v-stickykeys) [inline grafana/provisioning/dashboards/dev/pubsub.json]: spacing is different. i don't mind changing the spacing but let's do that is separate prs to make the diff easier to read.

would be even better to add prettier if you don't mind

File: grafana/provisioning/dashboards/dev/pubsub.json [hunk 2]
```diff
@@ -1,1576 +1,1576 @@
...
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "alias": "update",
+          "dimensions": {},
+          "expression": "",
+          "id": "",
+          "matchExact": true,
+          "metricName": "pubsub-elp-1-1-snd-upd",
```
Comment (v-stickykeys) [inline grafana/provisioning/dashboards/dev/pubsub.json]: dev does not have elp nodes so panels for that should not be here

**Review by v-stickykeys — CHANGES_REQUESTED** [review]
1. Can you add screenshots of before and after? That way it will be easier to know what has changed at least visually
2. Dev and Tnet should have panels for a public r/w node, prod should not
3. Only prod should have panels for elp nodes

File: grafana/provisioning/dashboards/dev/pubsub.json (hunk 1 above)
Comment (gvelez17) [reply grafana/provisioning/dashboards/dev/pubsub.json]: that was it!  i should have seen that - adjusted the spacing to match and now we see only actual differences.

Not sure if I should keep the change from 'Cloudwatch' to 'Cloudwatch-1'

Verified that the re-upload of the new (exported and spacing adjusted) dashboards does match what is presently on the site.

Also verified that the dashboards in dev match what was on the site prior to the gateway changes.

It seems that the change to cloudwatch-1 is not significant?

**Comment by gvelez17** [comment]
I can generate and attach the screenshots, will do that but also there is a test failure?

File: grafana/provisioning/dashboards/dev/pubsub.json (hunk 2 above)
Comment (gvelez17) [reply grafana/provisioning/dashboards/dev/pubsub.json]: removed

**Comment by gvelez17** [comment]
Hm @v-stickykeys  do we still need before/after shots as now the graphs are nearly identical except the requested changes?  The thing is if I take screenshots before deploy I'm actually screenshotting my UI changes, not the deployed changes in the PR.  So it might be misleading, better let the code speak for itself?

On the other hand its a good idea to have a visual review - just note that the updated versions are the UI changes that were then exported and put in this PR, not generated from this PR

**Comment by gvelez17** [comment]
prod pubsub gateway existing vs updated

image

image

**Comment by gvelez17** [comment]
Clay original (note incorrectly identical dashboards):

image

Clay after PR:

image

**Comment by gvelez17** [comment]
original dev

image

updated dev (nearly, but not exactly identical now)

image

File: grafana/provisioning/dashboards/prod/pubsub.json [hunk 3]
```diff
@@ -16,15 +16,15 @@
...
+    "iteration": 1649366630728,
     "links": [],
     "panels": [
         {
             "aliasColors": {},
             "bars": false,
             "dashLength": 10,
             "dashes": false,
-            "datasource": "CloudWatch",
+            "datasource": "CloudWatch-1",
```
Comment (v-stickykeys) [inline grafana/provisioning/dashboards/prod/pubsub.json]: We should use only 1 datasource caled "CloudWatch" but need to add to its connection details the appropriate `Namespaces of Custom Metrics` field
image

**Review by v-stickykeys — CHANGES_REQUESTED** [review]
Need to update the config for cloudwatch datasource

**Comment by v-stickykeys** [comment]
The first set of screenshots say "QA-dev-version-of-PubSub" what does this mean?

File: grafana/provisioning/dashboards/prod/pubsub.json (hunk 3 above)
Comment (gvelez17) [reply grafana/provisioning/dashboards/prod/pubsub.json]: not sure where to enter this, I think the namespace is already in the json?

**Comment by gvelez17** [comment]
> The first set of screenshots say "QA-dev-version-of-PubSub" what does this mean?

Since I was taking the screenshots after I had already changed the dashboards in the UI, I had to upload temporary dashboards with the old json for comparison, I gave them temporary names starting with "QA" and then deleted them after screenshots

**Comment by gvelez17** [comment]
@v-stickykeys so I went thru the jsons again and actually found that I'd missed a few entries, on prod and clay I had not updated the qry/rsp/alv entries, fixed now.

However I'm not seeing where to update the namespace.  When I did the last update it saved as CloudWatch not CloudWatch-1 so that seems no longer an issue.  The updates now are from correcting again on the UI (after the deploy) and saving.

File: grafana/provisioning/dashboards/prod/pubsub.json (hunk 3 above)
Comment (v-stickykeys) [reply grafana/provisioning/dashboards/prod/pubsub.json]: Oh yeah have it here: https://github.com/3box/ceramic-stats/blob/dev/grafana/provisioning/datasources/datasource.yaml#L17

So as long as we are passing in the environment variable I think we should be good on this. Will just have to be different for each env (prod, tnet, dev)

And then can use "Cloudwatch" instead of Cloudwatch-1

**Comment by gvelez17** [comment]
we walked thru this live and addressed the comments, just the tests were not passing (needed trivial readme change to rerun)
